subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/AbstractFileCollectionFingerprinter.java

```diff
@@ -33,6 +35,8 @@
  */
 @NonNullApi
 public abstract class AbstractFileCollectionFingerprinter implements FileCollectionFingerprinter {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
```

Comment:
I used an instance field logger so the implementing class was used as the logger category, rather than the abstract class. The fingerprinter id is in the output though, so switching to a static logger isn't the end of the world.